core/src/test/scala/unit/kafka/server/epoch/util/ReplicaFetcherMockBlockingSend.scala

```diff
@@ -78,7 +80,19 @@ class ReplicaFetcherMockBlockingSend(offsets: java.util.Map[TopicPartition, Epoc
         callback.foreach(_.apply())
         epochFetchCount += 1
         lastUsedOffsetForLeaderEpochVersion = requestBuilder.latestAllowedVersion()
-        new OffsetsForLeaderEpochResponse(currentOffsets)
+
+        val data = new OffsetForLeaderEpochResponseData()
+        currentOffsets.forEach((tp, offsetForLeaderPartition) => {
+          var topic = data.topics.find(tp.topic)
+          if (topic == null) {
+            topic = new OffsetForLeaderTopicResult()
+              .setTopic(tp.topic)
+            data.topics.add(topic)
+          }
+          topic.partitions.add(offsetForLeaderPartition.setPartition(tp.partition))
```

Comment:
Should it be ```topic.partitions.add(offsetForLeaderPartition.duplicate())```?

Reply:
Hum.. I am not sure to understand the `duplicate` suggestion here. Could you elaborate?

Reply:
oh, it is just personal taste. the code ```topic.partitions.add(offsetForLeaderPartition.setPartition(tp.partition))``` adds inner mutable object (```offsetForLeaderPartition```) to returned response. I prefer to add a copy of ```offsetForLeaderPartition```.

Reply:
Good point. Actually, `setPartition(tp.partition)` is not needed. Let me remove it.